Sources/TuistCore/MetadataProviders/FrameworkMetadataReader.swift

```diff
@@ -0,0 +1,199 @@
+import Foundation
+import MachO
+import TSCBasic
+import TuistGraph
+/// FrameworkMetadataReader reads a framework/library metadata using the Mach-o file format.
+/// Useful documentation:
+/// - https://opensource.apple.com/source/cctools/cctools-809/misc/lipo.c
+/// - https://opensource.apple.com/source/xnu/xnu-4903.221.2/EXTERNAL_HEADERS/mach-o/loader.h.auto.html
+
+class FrameworkMetadataReader {
```

Comment:
This is awesome @adellibovi, thanks a lot for doing it. One small suggestion that I'd make is that I wouldn't create another class. If you notice the diff of `PrecompiledMetadataProvider`, you've removed all the logic from it and the methods delegate to `FrameworkMetadataReader`. Having an extra level of delegation adds unnecessary complexity so I'd rather place all the logic in `PrecompiledMetadataProvider`. What do you think?

Reply:
Looks good to me. Wasn't sure if you would prefer to have this parsing in a separate class. Will move it soon 🙏🏻